File: seeder-client/src/main/java/com/DTOs/CarDTO.java
package com.DTOs;

import com.Enums.EngineType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.*;

@Getter
@Setter
@AllArgsConstructor
public class CarDTO {
    private Long id;
    private String vin;
    private String Manufacturer;
    private String Model;
    private Integer Year;
    private EngineType type;
    private Date createdAt;
    private Collection<RepairDTO> repairments;
}

File: end-user-client/src/models/CarDto.ts
import { EngineType } from './enums/EngineType';
import RepairDto from './RepairDto';

export default interface CarDto {
    id: number;
    type: EngineType;
    createdAt: Date;
    repairments: RepairDto[];
    year: number;
    vin: string;
    manufacturer: string;
    model: string;
}
